test/e2e/node/pod_resize.go

```diff
@@ -0,0 +1,915 @@
+		c, found := cMap[ci.Name]
+		gomega.Expect(found == true)
+		tc, _ := makeTestContainer(ci)
+		framework.ExpectEqual(c.Resources, tc.Resources)
+	}
+}
+
+func verifyPodAllocations(pod *v1.Pod, tcInfo []TestContainerInfo) {
+	cStatusMap := make(map[string]*v1.ContainerStatus)
+	for i, c := range pod.Status.ContainerStatuses {
+		cStatusMap[c.Name] = &pod.Status.ContainerStatuses[i]
+	}
+
+	for _, ci := range tcInfo {
+		cStatus, found := cStatusMap[ci.Name]
+		gomega.Expect(found == true)
+		if ci.Allocations == nil {
+			alloc := &ContainerAllocations{CPUAlloc: ci.Resources.CPUReq, MemAlloc: ci.Resources.MemReq}
+			ci.Allocations = alloc
+			defer func() {
+				ci.Allocations = nil
+			}()
+		}
+
+		_, tcStatus := makeTestContainer(ci)
+		framework.ExpectEqual(cStatus.ResourcesAllocated, tcStatus.ResourcesAllocated)
+	}
+}
+
+func verifyPodStatusResources(pod *v1.Pod, tcInfo []TestContainerInfo) {
+	csMap := make(map[string]*v1.ContainerStatus)
+	for i, c := range pod.Status.ContainerStatuses {
+		csMap[c.Name] = &pod.Status.ContainerStatuses[i]
+	}
+	for _, ci := range tcInfo {
+		cs, found := csMap[ci.Name]
+		gomega.Expect(found == true)
+		tc, _ := makeTestContainer(ci)
+		framework.ExpectEqual(cs.Resources, tc.Resources)
+	}
+}
+
+func verifyPodContainersCgroupConfig(pod *v1.Pod, tcInfo []TestContainerInfo) {
+	verifyCgroupValue := func(cName, cgPath, expectedCgValue string) {
+		cmd := []string{"head", "-n", "1", cgPath}
+		cgValue, err := framework.LookForStringInPodExecToContainer(pod.Namespace, pod.Name, cName, cmd, expectedCgValue, PollTimeout)
+		framework.ExpectNoError(err, "failed to find expected cgroup value in container")
+		cgValue = strings.Trim(cgValue, "\n")
+		gomega.Expect(cgValue == expectedCgValue)
+	}
+	for _, ci := range tcInfo {
+		if ci.Resources == nil {
+			continue
+		}
+		tc, _ := makeTestContainer(ci)
+		if tc.Resources.Limits != nil || tc.Resources.Requests != nil {
+			var cpuShares int64
+			memLimitInBytes := tc.Resources.Limits.Memory().Value()
+			cpuRequest := tc.Resources.Requests.Cpu()
+			cpuLimit := tc.Resources.Limits.Cpu()
+			if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+				cpuShares = int64(kubecm.MilliCPUToShares(cpuLimit.MilliValue()))
+			} else {
+				cpuShares = int64(kubecm.MilliCPUToShares(cpuRequest.MilliValue()))
+			}
+			cpuQuota := kubecm.MilliCPUToQuota(cpuLimit.MilliValue(), kubecm.QuotaPeriod)
+			if cpuLimit.IsZero() {
+				cpuQuota = -1
+			}
+			verifyCgroupValue(ci.Name, CgroupCPUShares, strconv.FormatInt(cpuShares, 10))
+			verifyCgroupValue(ci.Name, CgroupCPUQuota, strconv.FormatInt(cpuQuota, 10))
+			if memLimitInBytes > 0 {
+				verifyCgroupValue(ci.Name, CgroupMemLimit, strconv.FormatInt(memLimitInBytes, 10))
+			}
+		}
+	}
+}
+
+func doPodResizeTest() {
+	f := framework.NewDefaultFramework("pod-resize")
+	var podClient *framework.PodClient
+	var ns string
+
+	if !utilfeature.DefaultFeatureGate.Enabled(InPlacePodVerticalScalingFeature) {
+		return
+	}
```

Comment:
What e2e suite is this running against? We're not setting the feature gate anywhere to ensure this can be tested and I don't see a selector for it.

Reply:
If I interpreted your question correctly. this is intended to run as part of [sig-node] (when featuregate is enabled in beta .. unlike unit tests, this requires binaries to be built with FG enabled)
```go
var _ = ginkgo.Describe("[sig-node] PodInPlaceResizeContainer", func() {
        doPodResizeTest()
})
```

The above code disable this test from running when featuregate is disabled in alpha, else it throws failures as resources are immutable. 

For my testing, I enable the featuregate and then run as follows:
```go
kubetest --test --test_args="--ginkgo.focus=PodInPlaceResize" --provider=local
```

Does that answer your question?

Reply:
You need to add e2es for this with the feature gate on, like e.g. https://github.com/kubernetes/test-infra/pull/22826

That can't be punted to beta.

Reply:
Please see latest commit. @wangchen615 fixed this and I've merged her commit. Thanks,